### ISSUE TITLE
[Bug]: Tuono dev crashes once with tuono dynamic

### ISSUE THREAD

**Issue by pveierland** [tuono-labs/tuono#216]
### Description

I tested both `React.lazy` and `tuono.dynamic` with tuono 0.16.0 as described.

### Dynamic component

```
// DynamicComponent.tsx
import { type JSX } from "react";

export const DynamicComponent = (): JSX.Element => {
	return <>DynamicComponent</>;
};

export default DynamicComponent;
```

### Test 1: `React.lazy`

```
// testReactLazy.tsx

import React, { Suspense, type JSX } from "react";

const DynamicComponent = React.lazy(() => import("../components/DynamicComponent"));

const LoadingComponent = (): JSX.Element => {
	console.log("Loading");
	return <>Loading</>;
};

export default function IndexPage(): JSX.Element | null {
	return (
		<Suspense fallback={<LoadingComponent />}>
			<DynamicComponent />
		</Suspense>
	);
}
```

This approach works with `tuono dev`, but results in the following client side warning/error logged in console:

```Uncaught Error: The server could not finish this Suspense boundary, likely due to an error during server rendering. Switched to client rendering.```

### Test 2: `tuono.dynamic`

```
// testTuonoDynamic.tsx

import { Suspense, type JSX } from "react";
import { dynamic } from "tuono";

const DynamicComponent: any = dynamic(() => import("../components/DynamicComponent"));

const LoadingComponent = (): JSX.Element => {
	console.log("Loading");
	return <>Loading</>;
};

export default function IndexPage(): JSX.Element | null {
	return (
		<Suspense fallback={<LoadingComponent />}>
			<DynamicComponent />
		</Suspense>
	);
}
```

This approach crashes `tuono dev` when starting the server, but after resetting it works, however with the same client side warning/error logged in console:

```Uncaught Error: The server could not finish this Suspense boundary, likely due to an error during server rendering. Switched to client rendering.```

The following is output by `tuono dev` upon start:

```
file:///project/node_modules/tuono-lazy-fn-vite-plugin/dist/esm/index.js:29
      if (init.callee.name === TUONO_DYNAMIC_FN_ID) {
                      ^

[vite-plugin-tuono-lazy-loading] unknown file: Cannot read properties of undefined (reading 'name')
file: /project/src/routes/fresh.tsx
    at file:///project/node_modules/tuono-lazy-fn-vite-plugin/dist/esm/index.js:29:23
    at Array.forEach (<anonymous>)
    at VariableDeclaration (file:///project/node_modules/tuono-lazy-fn-vite-plugin/dist/esm/index.js:27:28)
    at NodePath._call (/project/node_modules/@babel/traverse/lib/path/context.js:49:20)
    at NodePath.call (/project/node_modules/@babel/traverse/lib/path/context.js:39:18)
    at NodePath.visit (/project/node_modules/@babel/traverse/lib/path/context.js:88:31)
    at TraversalContext.visitQueue (/project/node_modules/@babel/traverse/lib/context.js:90:16)
    at TraversalContext.visitMultiple (/project/node_modules/@babel/traverse/lib/context.js:62:17)
    at TraversalContext.visit (/project/node_modules/@babel/traverse/lib/context.js:111:19)
    at traverseNode (/project/node_modules/@babel/traverse/lib/traverse-node.js:22:17)
    at traverse (/project/node_modules/@babel/traverse/lib/index.js:53:34)
    at NodePath.traverse (/project/node_modules/@babel/traverse/lib/path/index.js:120:24)
    at ImportSpecifier (file:///project/node_modules/tuono-lazy-fn-vite-plugin/dist/esm/index.js:50:18)
    at NodePath._call (/project/node_modules/@babel/traverse/lib/path/context.js:49:20)
    at NodePath.call (/project/node_modules/@babel/traverse/lib/path/context.js:39:18)
    at NodePath.visit (/project/node_modules/@babel/traverse/lib/path/context.js:88:31)
    at TraversalContext.visitQueue (/project/node_modules/@babel/traverse/lib/context.js:90:16)
    at TraversalContext.visitMultiple (/project/node_modules/@babel/traverse/lib/context.js:62:17)
    at TraversalContext.visit (/project/node_modules/@babel/traverse/lib/context.js:111:19)
    at traverseNode (/project/node_modules/@babel/traverse/lib/traverse-node.js:22:17)
    at NodePath.visit (/project/node_modules/@babel/traverse/lib/path/context.js:94:52)
    at TraversalContext.visitQueue (/project/node_modules/@babel/traverse/lib/context.js:90:16)
    at TraversalContext.visitMultiple (/project/node_modules/@babel/traverse/lib/context.js:62:17)
    at TraversalContext.visit (/project/node_modules/@babel/traverse/lib/context.js:111:19)
    at traverseNode (/project/node_modules/@babel/traverse/lib/traverse-node.js:22:17)
    at traverse (/project/node_modules/@babel/traverse/lib/index.js:53:34)
    at NodePath.traverse (/project/node_modules/@babel/traverse/lib/path/index.js:120:24)
    at PluginPass.Program (file:///project/node_modules/tuono-lazy-fn-vite-plugin/dist/esm/index.js:47:12)
    at newFn (/project/node_modules/@babel/traverse/lib/visitors.js:172:14)
    at NodePath._call (/project/node_modules/@babel/traverse/lib/path/context.js:49:20)
    at NodePath.call (/project/node_modules/@babel/traverse/lib/path/context.js:39:18)
    at NodePath.visit (/project/node_modules/@babel/traverse/lib/path/context.js:88:31)
    at TraversalContext.visitQueue (/project/node_modules/@babel/traverse/lib/context.js:90:16)
    at TraversalContext.visitSingle (/project/node_modules/@babel/traverse/lib/context.js:66:19)
    at TraversalContext.visit (/project/node_modules/@babel/traverse/lib/context.js:113:19)
    at traverseNode (/project/node_modules/@babel/traverse/lib/traverse-node.js:22:17)
    at traverse (/project/node_modules/@babel/traverse/lib/index.js:53:34)
    at transformFile (/project/node_modules/@babel/core/lib/transformation/index.js:80:31)
    at transformFile.next (<anonymous>)
    at run (/project/node_modules/@babel/core/lib/transformation/index.js:25:12)
    at run.next (<anonymous>)
    at transform (/project/node_modules/@babel/core/lib/transform.js:22:33)
    at transform.next (<anonymous>)
    at evaluateSync (/project/node_modules/gensync/index.js:251:28)
    at sync (/project/node_modules/gensync/index.js:89:14)
    at stopHiding - secret - don't use this - v1 (/project/node_modules/@babel/core/lib/errors/rewrite-stack-trace.js:47:12)
    at Module.transformSync (/project/node_modules/@babel/core/lib/transform.js:42:76)
    at Object.transform (file:///project/node_modules/tuono-lazy-fn-vite-plugin/dist/esm/index.js:63:27)
    at Object.handler (file:///project/node_modules/tuono/node_modules/vite/dist/node/chunks/dep-CB_7IfJ-.js:65683:15)
    at file:///project/node_modules/rollup/dist/es/shared/node-entry.js:20812:40 {
  code: 'PLUGIN_ERROR',
  pluginCode: 'BABEL_TRANSFORM_ERROR',
  plugin: 'vite-plugin-tuono-lazy-loading',
  hook: 'transform',
  id: '/project/src/routes/fresh.tsx',
  watchFiles: [
    '/project/.tuono/server-main.tsx',
    '/project/node_modules/tuono/dist/esm/ssr/index.js',
    '/project/.tuono/routeTree.gen.ts',
    '/project/node_modules/react/jsx-runtime.js',
    '/project/node_modules/fast-text-encoding/text.min.js',
    '/project/node_modules/react-dom/server.browser.js',
    '/project/node_modules/react-helmet-async/lib/index.esm.js',
    '/project/node_modules/tuono-router/dist/esm/index.js',
    '/project/package.json',
    '/project/node_modules/tuono/dist/esm/index.js',
    '/project/src/routes/__root.tsx',
    '/project/src/routes/fresh.tsx'
  ]
}

Node.js v20.17.0

  ⚡ Tuono v0.16.0
  Ready at: http://localhost:3000

  GET /fresh 200 in 32ms
  GET /installHook.js.map 200 in 50ms
  GET /installHook.js.map 200 in 34ms
```

### Expected behaviour

- It is expected that no ```Uncaught Error: The server could not finish this Suspense boundary, likely due to an error during server rendering. Switched to client rendering.``` warning is shown if possible.
- It is expected that `tuono dev` does not crash when started with `tuono.dynamic` use.

### How to reproduce

- Add the described components.
- Run `tuono dev`.

### Tuono version

0.16.0

### OS

NixOS

### Browser

Firefox

### Node version

v20.17.0

### Rust version

rustc 1.82.0 (f6e511eec 2024-10-15)

### Crate version

0.16.0

### Node Package Manger version

Yarn 1.22.22

### Additional context

Note: It could be helpful to make it clear that `lazyLoadComponent` is an internal function and should not be used.

**Comment by Valerioageno** [comment]
Logging here the initial investigation:
1. `Uncaught Error: The server could not finish this Suspense boundary,` This error seems to be due to the development react bundler. Still need to find the root cause, but it gets correctly rendered and stripped out on the production build
2. `unknown file: Cannot read properties of undefined (reading 'name')` This error appear when `dynamic` is the only function exported by the `tuono` library `import { dynamic } from 'tuono'`. We need to improve the internal vite plugin + adding this test case
3. `lazyLoadComponent` should be renamed. I'd propose something like `tuono__internal__lazyLoadComponent`

**Comment by Valerioageno** [comment]
Update:
1. This issue is related to the way we are server side rendering the project. We are addressing this, but it will take longer than expected (you can use suspense, but you will have that log on the console in the development mode).
2. This is fixed here: https://github.com/tuono-labs/tuono/pull/223
3. This is fixed here: https://github.com/tuono-labs/tuono/pull/222

**Comment by Valerioageno** [comment]
The last issue has been fixed with `tuono v0.16.5`. Please let me know if something still does not work
